src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -5874,6 +5883,66 @@ private ScanTypeFlags ScanTupleType(out SyntaxToken lastTokenOfType)
             return ScanTypeFlags.NotType;
         }
 
+#nullable enable
+        private ScanTypeFlags ScanFunctionPointerType(out SyntaxToken lastTokenOfType)
+        {
+            Debug.Assert(IsFunctionPointerStart());
+            _ = EatContextualToken(SyntaxKind.DelegateKeyword);
+            lastTokenOfType = EatToken(SyntaxKind.AsteriskToken);
+
+            if (IsPossibleCallingConvention())
+            {
+                _ = EatToken();
+            }
+
+            if (!IsPossibleFunctionPointerParameterListStart(CurrentToken))
+            {
+                return ScanTypeFlags.MustBeType;
+            }
+
+            _ = EatToken(SyntaxKind.LessThanToken);
+
+            var saveTerm = _termState;
+            _termState |= TerminatorState.IsEndOfFunctionPointerParameterList;
+
+            try
+            {
+                do
+                {
+                    while (IsFunctionPointerParameterModifier())
+                    {
+                        _ = EatToken();
+                    }
+
+                    _ = ScanType(out _);
+
+                    if (skipBadFunctionPointerParameterTokens() == PostSkipAction.Abort)
+                    {
+                        break;
+                    }
+
+                    _ = EatToken(SyntaxKind.CommaToken);
+                }
+                while (true);
+            }
+            finally
+            {
+                _termState = saveTerm;
+            }
+
+            lastTokenOfType = EatToken(SyntaxKind.GreaterThanToken);
```

Comment:
>EatToken(SyntaxKind.GreaterThanToken) [](start = 30, length = 37)

Are we really interested in "fake"/missing tokens returned as ```lastTokenOfType```?

Reply:
It looks like the only thing that anyone is interested in the `lastTokenOfType` for is to know whether it's a predefined type, which will of course be false for this scenario.

---
In reply to: [349369399](https://github.com/dotnet/roslyn/pull/39879#discussion_r349369399) [](ancestors = 349369399)

Reply:
And a couple of uses to ensure we don't parse `Type??`

---
In reply to: [349370219](https://github.com/dotnet/roslyn/pull/39879#discussion_r349370219) [](ancestors = 349370219,349369399)

Reply:
> It looks like the only thing that anyone is interested in the lastTokenOfType for is to know whether it's a predefined type, which will of course be false for this scenario.

It would be more robust to not rely on the **current** usage elsewhere and sticking only to real tokens in the stream. 

---
In reply to: [349370569](https://github.com/dotnet/roslyn/pull/39879#discussion_r349370569) [](ancestors = 349370569,349370219,349369399)